From: Michael Froman <mjfroman@mac.com>
Date: Wed, 20 Mar 2024 11:32:05 -0500
Subject: Bug 1886497 - Cherry-pick upstream libwebrtc commit de3c726121
 r?dbaker

Upstream commit: https://webrtc.googlesource.com/src/+/de3c726121bd097e79a8f1aa43df48aab7e2237f
       Update to vpython 3.11 and remove .vpython (v2.x)

       Bug: b/310806212
       Change-Id: I7fdb12ee4f83410bed9358e7249e4601e773056f
       Reviewed-on: https://webrtc-review.googlesource.com/c/src/+/335641
       Reviewed-by: Mirko Bonadei <mbonadei@webrtc.org>
       Commit-Queue: Christoffer Dewerin <jansson@google.com>
       Cr-Commit-Position: refs/heads/main@{#41607}
---
 .vpython  | 76 -------------------------------------------------------
 .vpython3 | 38 ++++++++++++++--------------
 2 files changed, 19 insertions(+), 95 deletions(-)
 delete mode 100644 .vpython

diff --git a/.vpython b/.vpython
deleted file mode 100644
index d226875f02..0000000000
--- a/.vpython
+++ /dev/null
@@ -1,76 +0,0 @@
-# This is a vpython "spec" file.
-#
-# It describes patterns for python wheel dependencies of the python scripts in
-# the chromium repo, particularly for dependencies that have compiled components
-# (since pure-python dependencies can be easily vendored into third_party).
-#
-# When vpython is invoked, it finds this file and builds a python VirtualEnv,
-# containing all of the dependencies described in this file, fetching them from
-# CIPD (the "Chrome Infrastructure Package Deployer" service). Unlike `pip`,
-# this never requires the end-user machine to have a working python extension
-# compilation environment. All of these packages are built using:
-#   https://chromium.googlesource.com/infra/infra/+/main/infra/tools/dockerbuild/
-#
-# All python scripts in the repo share this same spec, to avoid dependency
-# fragmentation.
-#
-# If you have depot_tools installed in your $PATH, you can invoke python scripts
-# in this repo by running them as you normally would run them, except
-# substituting `vpython` instead of `python` on the command line, e.g.:
-#   vpython path/to/script.py some --arguments
-#
-# Read more about `vpython` and how to modify this file here:
-#   https://chromium.googlesource.com/infra/infra/+/main/doc/users/vpython.md
-
-python_version: "2.7"
-
-# Used by:
-#   third_party/catapult
-wheel: <
-  name: "infra/python/wheels/psutil/${platform}_${py_python}_${py_abi}"
-  version: "version:5.2.2"
->
-
-# Used by tools_webrtc/perf/process_perf_results.py.
-wheel: <
-  name: "infra/python/wheels/httplib2-py2_py3"
-  version: "version:0.10.3"
->
-
-# Used by:
-#   build/toolchain/win
-wheel: <
-  name: "infra/python/wheels/pypiwin32/${vpython_platform}"
-  version: "version:219"
-  match_tag: <
-    platform: "win32"
-  >
-  match_tag: <
-    platform: "win_amd64"
-  >
->
-
-wheel: <
-  name: "infra/python/wheels/six-py2_py3"
-  version: "version:1.15.0"
->
-wheel: <
-  name: "infra/python/wheels/pbr-py2_py3"
-  version: "version:3.0.0"
->
-wheel: <
-  name: "infra/python/wheels/funcsigs-py2_py3"
-  version: "version:1.0.2"
->
-wheel: <
-  name: "infra/python/wheels/mock-py2_py3"
-  version: "version:2.0.0"
->
-wheel: <
-  name: "infra/python/wheels/protobuf-py2_py3"
-  version: "version:3.13.0"
->
-wheel: <
-  name: "infra/python/wheels/requests-py2_py3"
-  version: "version:2.13.0"
->
diff --git a/.vpython3 b/.vpython3
index 3f571df261..2be8efaa0a 100644
--- a/.vpython3
+++ b/.vpython3
@@ -22,24 +22,24 @@
 # Read more about `vpython` and how to modify this file here:
 #   https://chromium.googlesource.com/infra/infra/+/main/doc/users/vpython.md
 
-python_version: "3.8"
+python_version: "3.11"
 
 # Used by:
 #   third_party/catapult
 wheel: <
   name: "infra/python/wheels/psutil/${vpython_platform}"
-  version: "version:5.8.0.chromium.2"
+  version: "version:5.8.0.chromium.3"
 >
 
 # Used by tools_webrtc/perf/process_perf_results.py.
 wheel: <
   name: "infra/python/wheels/httplib2-py3"
-  version: "version:0.19.1"
+  version: "version:0.22.0"
 >
 
 wheel: <
-  name: "infra/python/wheels/pyparsing-py2_py3"
-  version: "version:2.4.7"
+  name: "infra/python/wheels/pyparsing-py3"
+  version: "version:3.1.1"
 >
 
 
@@ -47,7 +47,7 @@ wheel: <
 #   build/toolchain/win
 wheel: <
   name: "infra/python/wheels/pywin32/${vpython_platform}"
-  version: "version:300"
+  version: "version:306"
   match_tag: <
     platform: "win32"
   >
@@ -59,48 +59,48 @@ wheel: <
 # GRPC used by iOS test.
 wheel: <
   name: "infra/python/wheels/grpcio/${vpython_platform}"
-  version: "version:1.44.0"
+  version: "version:1.57.0"
 >
 
 wheel: <
   name: "infra/python/wheels/six-py2_py3"
-  version: "version:1.15.0"
+  version: "version:1.16.0"
 >
 wheel: <
   name: "infra/python/wheels/pbr-py2_py3"
-  version: "version:3.0.0"
+  version: "version:5.9.0"
 >
 wheel: <
   name: "infra/python/wheels/funcsigs-py2_py3"
   version: "version:1.0.2"
 >
 wheel: <
-  name: "infra/python/wheels/mock-py2_py3"
-  version: "version:2.0.0"
+  name: "infra/python/wheels/mock-py3"
+  version: "version:4.0.3"
 >
 wheel: <
   name: "infra/python/wheels/protobuf-py3"
-  version: "version:3.20.0"
+  version: "version:4.25.1"
 >
 wheel: <
   name: "infra/python/wheels/requests-py3"
   version: "version:2.31.0"
 >
 wheel: <
-  name: "infra/python/wheels/idna-py2_py3"
-  version: "version:2.8"
+  name: "infra/python/wheels/idna-py3"
+  version: "version:3.4"
 >
 wheel: <
-  name: "infra/python/wheels/urllib3-py2_py3"
-  version: "version:1.26.6"
+  name: "infra/python/wheels/urllib3-py3"
+  version: "version:2.1.0"
 >
 wheel: <
-  name: "infra/python/wheels/certifi-py2_py3"
-  version: "version:2020.11.8"
+  name: "infra/python/wheels/certifi-py3"
+  version: "version:2023.11.17"
 >
 wheel: <
   name: "infra/python/wheels/charset_normalizer-py3"
-  version: "version:2.0.4"
+  version: "version:3.3.2"
 >
 wheel: <
   name: "infra/python/wheels/brotli/${vpython_platform}"
